util/Migrator/DbScripts/2023-09-14_00_Remove_WebAuthn_For_Free_Users.sql

```diff
@@ -0,0 +1,39 @@
+-- Assumptions:
+-- When a 2FA method is disabled, it is removed from the TwoFactorProviders array
+
+-- Problem statement:
+-- We have users who currently do not have any 2FA method, with the only one being
+-- WebAuthn, which is effectively disabled by a server-side permission check for Premium status.
+-- With WebAuthn being made free, we want to avoid these users suddenly being forced
+-- to provide 2FA using a key that they haven't used in a long time, by deleting that key from their TwoFactorProviders.
+declare @TwoFactorByUser TABLE
+(
+    Id UNIQUEIDENTIFIER,
+    Email NVARCHAR(256),
+    TwoFactorType INT
+)
+
+INSERT INTO @TwoFactorByUser
+SELECT  u.Id, 
+        u.Email,
+        tfp1.[key] as TwoFactorType
+FROM [User] u
+LEFT OUTER JOIN [OrganizationUser] ou on ou.UserId = u.Id
+LEFT OUTER JOIN [Organization] o on o.Id = ou.OrganizationId
+CROSS APPLY OPENJSON(u.TwoFactorProviders) tfp1
+CROSS APPLY OPENJSON(tfp1.[value]) WITH (
+   [Enabled] nvarchar(10) '$.Enabled'
+) tfp2
+WHERE [Enabled] = 'true' -- We only want enabled 2FA methods
+AND Premium = 0 -- User isn't Premium
+AND (o IS NULL OR (o IS NOT NULL AND o.Enabled = 1 AND o.UsersGetPremium = 0)) -- User doesn't have Premium from their org
```

Comment:
🎨 Since we're running this as a script we have the benefit of time and an ad hoc run, so why not make this less complicated? I think it'd be cleaner and with more straightforward results if we ran the above twice for users in or out of an org. This looks fine to me regardless.

Reply:
I changed the approach here with a new commit - instead pre-selecting all the users without Premium, and then using that as the base for the `CROSS APPLY`.  This simplifies this logic and perhaps also reduces the number of rows for which the `CROSS APPLY` needs to work.